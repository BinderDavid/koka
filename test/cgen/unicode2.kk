val s0 = "ma\xF1ana"     // single character
val s1 = "man\u0303ana"  // combining character
val s2 = "x\u0307\u0323xyz" // combining mark

fun test1(s : string,ofs = 1) {
  println( "len: " + s.length.show + 
          ", sub3: " + s.substr(ofs,3) + 
          ", rev3: " + s.substr(~5,3) +
          ",\n  s: " + s +
          ", " + s.list.show-list(fun(i){show-hex(i.int)}) )
}

fun main() {
  s0.test1
  s1.test1
  s2.test1
}
