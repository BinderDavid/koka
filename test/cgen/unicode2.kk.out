len: 6, sub3: aña, rev3: aña,
 s: mañana, [6D,61,F1,61,6E,61]
len: 7, sub3: añ, rev3: ña,
 s: mañana, [6D,61,6E,303,61,6E,61]
len: 6, sub3: ̣̇x, rev3: ̣̇x,
 s: ẋ̣xyz, [78,307,323,78,79,7A]